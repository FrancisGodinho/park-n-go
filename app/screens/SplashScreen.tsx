import { SafeAreaView, StyleSheet, Text, View } from "react-native";
import React from "react";
import { FontAwesome5 } from "@expo/vector-icons";
import Colors from "../constants/Colors";

type Props = {};

const SplashScreen = () => {
  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.logo}>
        <FontAwesome5 name="car" size={40} color={Colors.black} />
      </View>
      <Text>Parking Bot</Text>
    </SafeAreaView>
  );
};

export default SplashScreen;

const styles = StyleSheet.create({
  container: { flex: 1, justifyContent: "center", alignItems: "center" },
  logo: {
    width: 200,
    height: 200,
    borderRadius: 100,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: Colors.primary,
  },
});
