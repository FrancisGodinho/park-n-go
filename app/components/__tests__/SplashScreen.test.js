import React from "react";
import renderer from "react-test-renderer";
import { cleanup } from "@testing-library/react-native";
import { fireEvent, render } from "@testing-library/react";
import SplashScreen from "../../screens/SplashScreen";

let findTextElement = function (tree, element) {
  console.warn(tree);
  return true;
};

describe("<SplashScreen />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SplashScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
