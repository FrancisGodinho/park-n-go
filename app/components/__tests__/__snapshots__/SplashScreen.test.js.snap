// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashScreen /> renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#66A182",
        "borderRadius": 100,
        "height": 200,
        "justifyContent": "center",
        "width": 200,
      }
    }
  >
    <Text />
  </View>
  <Text>
    Parking Bot
  </Text>
</RCTSafeAreaView>
`;
